engine/gamesys/src/dmsdk/gamesys/components/comp_collection_factory.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020-2025 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#ifndef DMSDK_GAMESYS_COLLECTION_FACTORY_H
+#define DMSDK_GAMESYS_COLLECTION_FACTORY_H
+
+#include <stdint.h>
+#include <dmsdk/dlib/hash.h>
+#include <dmsdk/dlib/vmath.h>
+#include <dmsdk/dlib/hashtable.h>
+
+namespace dmGameObject
+{
+    typedef struct CollectionHandle* HCollection;
+    typedef dmHashTable<dmhash_t, HPropertyContainer> InstancePropertyBuffers;
+    typedef dmHashTable<dmhash_t, dmhash_t> InstanceIdMap;
+}
```

Comment:
While I'm all for forward declarations, it feels strange to not have them public as well.
Please add these to `engine/gameobject/src/dmsdk/gameobject/gameobject.h`.